serving/resource_model_serving.go

```diff
@@ -13,6 +14,85 @@ import (
 const DefaultProvisionTimeout = 45 * time.Minute
 const deleteCallTimeout = 10 * time.Second
 
+// updateConfig updates the configuration of the provided serving endpoint to the provided config.
+func updateConfig(ctx context.Context, w *databricks.WorkspaceClient, name string, e *serving.EndpointCoreConfigInput, d *schema.ResourceData) error {
+	e.Name = name
+	waiter, err := w.ServingEndpoints.UpdateConfig(ctx, *e)
+	if err != nil {
+		return err
+	}
+	_, err = waiter.GetWithTimeout(d.Timeout(schema.TimeoutUpdate))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// updateTags updates the tags of the provided serving endpoint to the given tags. Any tags not present on the existing
+// endpoint will be removed, any tags absent on the endpoint will be added, existing tags will be updated, and unchanged
+// tags will remain as-is.
+func updateTags(ctx context.Context, w *databricks.WorkspaceClient, name string, newTags []serving.EndpointTag, d *schema.ResourceData) error {
+	currentEndpoint, err := w.ServingEndpoints.Get(ctx, serving.GetServingEndpointRequest{
+		Name: name,
+	})
+	oldTags := currentEndpoint.Tags
+	if err != nil {
+		return err
+	}
+	req := serving.PatchServingEndpointTags{
+		Name: name,
+	}
+	for _, newTag := range newTags {
+		found := false
+		for _, oldTag := range oldTags {
+			if oldTag.Key == newTag.Key && oldTag.Value == newTag.Value {
+				found = true
+				break
+			}
+		}
+		if !found {
+			req.AddTags = append(req.AddTags, newTag)
+		}
+	}
+	for _, oldTag := range oldTags {
+		found := false
+		for _, newTag := range newTags {
+			if oldTag.Key == newTag.Key {
+				found = true
+				break
+			}
+		}
+		if !found {
+			req.DeleteTags = append(req.DeleteTags, oldTag.Key)
+		}
+	}
+	if _, err := w.ServingEndpoints.Patch(ctx, req); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Update the rate limit configuration for a model serving endpoint.
+func updateRateLimits(ctx context.Context, w *databricks.WorkspaceClient, name string, newRateLimits []serving.RateLimit, d *schema.ResourceData) error {
```

Comment:
this endpoint is now deprecated, you can remove it
`This API is deprecated for your Foundation Model endpoint. Please use AI Gateway to manage rate limits.`

Reply:
@mgyucht - see Vuong's comment